armi/operators/tests/test_operators.py

```diff
@@ -103,37 +111,96 @@ def test_loadStateError(self):
 
-            # test cases where no coupling interaction due to settings
-            self.r.core.p.coupledIteration = 0
-            noCouplingInteractions = 0
-            # because cyclesSkipTightCouplingInteraction above
+    def test_performTightCoupling_skip(self):
+        """ensure that cycles within cs[CONF_CYCLES_SKIP_TIGHT_COUPLING_INTERACTION] are skipped"""
+        self.o.cs[CONF_CYCLES_SKIP_TIGHT_COUPLING_INTERACTION] = [1]
+        with mockRunLogs.BufferLog() as mock:
             self.o._performTightCoupling(1, 0, writeDB=False)
-            self.assertEqual(self.r.core.p.coupledIteration, noCouplingInteractions)
-            self.o.cs["tightCoupling"] = False
-            self.o._performTightCoupling(2, 0, writeDB=False)
-            self.assertEqual(self.r.core.p.coupledIteration, noCouplingInteractions)
+            self.assertIn("interactAllCoupled disabled this cycle", mock.getStdout())
+            self.assertEqual(self.o.r.core.p.coupledIteration, 0)
+
+    def test_performTightCoupling_notConverged(self):
+        """ensure that the appropriate runLog.warning is addressed in tight coupling reaches max num of iters"""
+
+        class NoConverge(TightCoupler):
+            def isConverged(self, _val: TightCoupler._SUPPORTED_TYPES) -> bool:
+                return False
+
+        # pylint: disable=abstract-method
+        class InterfaceNoConverge(Interface):
+            name = "NoConverge"
+
+            def __init__(self, r, cs):
+                super().__init__(r, cs)
+                self.coupler = NoConverge(param="dummy", tolerance=None, maxIters=1)
+
+            def getTightCouplingValue(self):
+                return 0.0
+
+        self.o.addInterface(InterfaceNoConverge(None, self.o.cs))
+        with mockRunLogs.BufferLog() as mock:
+            self.o._performTightCoupling(0, 0, writeDB=False)
+            self.assertIn(
+                "have not converged! The maximum number of iterations", mock.getStdout()
+            )
+
+    def test_performTightCoupling_WriteDB(self):
+        """ensure a tight coupling iteration accours and that a DB WILL be written if requested"""
+        hasCouplingInteraction = 1
+        with directoryChangers.TemporaryDirectoryChanger():
+            with mockRunLogs.BufferLog() as mock:
+                self.dbWriteForCoupling(writeDB=True)
+                self.assertIn("Writing to database for statepoint:", mock.getStdout())
+                self.assertEqual(
+                    self.o.r.core.p.coupledIteration, hasCouplingInteraction
+                )
+
+    def test_performTightCoupling_NoWriteDB(self):
+        """ensure a tight coupling iteration accours and that a DB WILL NOT be written if requested"""
+        hasCouplingInteraction = 1
+        with directoryChangers.TemporaryDirectoryChanger():
+            self.dbWriteForCoupling(writeDB=False)
+            self.assertEqual(self.o.r.core.p.coupledIteration, hasCouplingInteraction)
```

Comment:
```suggestion
            with mockRunLogs.BufferLog() as mock:
                self.dbWriteForCoupling(writeDB=False)
                self.assertNotIn("Writing to database for statepoint:", mock.getStdout())
                self.assertEqual(self.o.r.core.p.coupledIteration, hasCouplingInteraction)
```

This is super optional...I just wanted to see some indication the db wasn't written